### PR TITLE
Cross-compile in CI

### PR DESCRIPTION
**Pull request by webrouse**
Jira: https://keboola.atlassian.net/browse/COM-817

**We create a new release in GitHub:**
image

**And after corresponding GitHub Action -> archives for all architectures are available for download:**
image

Test release: https://github.com/keboola/keboola-as-code/releases/tag/v0.0.6
Test GH Action: https://github.com/keboola/keboola-as-code/actions/runs/885991639

**You can download archive for your architecture -> and binary should print `Hello World!`.**

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -9,3 +9,31 @@ jobs:
...
+      - name: Set version
+        run: |
+          if [[ "${GITHUB_REF}" == refs/tags/* ]];
+          then
+            export TARGET_VERSION=${GITHUB_REF/refs\/tags\//}
+          else
+            export TARGET_VERSION="dev"
+          fi
+          echo ${TARGET_VERSION}
+          echo "TARGET_VERSION=${TARGET_VERSION}" >> $GITHUB_ENV
```
Comment (webrouse) [inline .github/workflows/push.yml:21]: Nastavi ENV premennu pre nasledujuce jobs.

File: cross-compile.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    GOARCH="${platform_split[1]}"
+    binary_path="./target/bin/${GOOS}/${GOARCH}/${binary_name}"
+    archive_path="./target/archive/${archive_name}-${TARGET_VERSION}-${GOOS}_${GOARCH}.zip"
+    if [ $GOOS = "windows" ]; then
+        binary_path+='.exe'
+    fi
+
+    # Compile
+    echo -ne "Compiling for ${platform} ... "
+    env GOOS=$GOOS GOARCH=$GOARCH CGO_ENABLED=0 go build -ldflags="-s -w" -o $binary_path ./src/main.go
```
Comment (webrouse) [inline cross-compile.sh:42]: Skompilujeme binarku pre danu architekturu.

File: cross-compile.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    if [ $? -ne 0 ]; then
+        echo 'An error has occurred!'
+        exit 1
+    fi
+
+    # Zip
+    zip -j -q "${archive_path}" "${binary_path}"
```
Comment (webrouse) [inline cross-compile.sh:49]: A spravime archiv.

**Review by ondrajodas — APPROVED** [review]
👍

**Pull request merged**